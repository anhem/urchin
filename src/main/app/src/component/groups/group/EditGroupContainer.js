import React, {Component} from 'react'
import {connect} from 'react-redux'
import EditGroup from './EditGroup';
import {addUser, deleteGroup, getGroup, getUsersForGroup, removeUser, setGroup} from '../../../action/groupAction';

class EditGroupContainer extends Component {

    componentWillMount() {
        this.props.getGroup(this.getGroupId());
        this.props.getUsersForGroup(this.getGroupId());
    }

    setGroup = (group) => {
        this.props.setGroup(group);
    };

    deleteGroup = (groupId) => {
        this.props.deleteGroup(groupId);
    };

    getGroupId = () => {
        return parseInt(this.props.match.params.id, 10);
    };

    render() {
        let groupId = this.getGroupId();
        return (
            <EditGroup
                groupId={groupId}
                group={this.props.group}
                usersInGroup={this.props.usersInGroup}
                fieldErrors={this.props.fieldErrors}
                callbacks={{
                    setGroup: this.setGroup,
                    deleteGroup: this.deleteGroup,
                    addUser: this.props.addUser,
                    removeUser: this.props.removeUser,
                }}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getGroup: (groupId) => {
            dispatch(getGroup(groupId))
        },
        setGroup: (group) => {
            dispatch(setGroup(group))
        },
        deleteGroup: (groupId) => {
            dispatch(deleteGroup(groupId))
        },
        getUsersForGroup: (groupId) => {
            dispatch(getUsersForGroup(groupId))
        },
        addUser: (groupId, userId) => {
            dispatch(addUser(groupId, userId))
        },
        removeUser: (groupId, userId) => {
            dispatch(removeUser(groupId, userId))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        group: state.groupData.group || {},
        usersInGroup: state.groupData.usersInGroup || [],
        fieldErrors: state.groupData.fieldErrors || {}
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditGroupContainer)