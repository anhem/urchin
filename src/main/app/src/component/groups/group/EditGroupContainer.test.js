import React from 'react';
import {mount} from 'enzyme'
import {EditGroupContainer} from "./EditGroupContainer";

describe('EditGroupContainer', () => {

    it('renders without crashing', () => {
        let props = {
            group: {},
            match: {
                params: {
                    id: 0
                }
            },
            usersInGroup: [],
            users: [],
            getGroup: jest.fn(),
            getUsersForGroup: jest.fn(),
            getUsers: jest.fn(),
        };

        expect(mount(<EditGroupContainer {...props}/>).length).toEqual(1);
    });
});
