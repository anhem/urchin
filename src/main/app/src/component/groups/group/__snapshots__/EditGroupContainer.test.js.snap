// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGroupContainer match snapshot 1`] = `
<EditGroup
  availableUsers={Array []}
  callbacks={
    Object {
      "addUser": undefined,
      "deleteGroup": undefined,
      "removeUser": undefined,
      "setGroup": undefined,
    }
  }
  group={Object {}}
  usersInGroup={Array []}
/>
`;
